File: token-validation-spring-test/src/main/java/no/nav/security/token/support/spring/test/EnableMockOAuth2Server.java
package no.nav.security.token.support.spring.test;

import org.springframework.boot.test.autoconfigure.properties.PropertyMapping;
import org.springframework.context.annotation.Import;

import java.lang.annotation.*;

@Documented
@Inherited
@Retention(RetentionPolicy.RUNTIME)
@Target(ElementType.TYPE)
@Import({
    MockOAuth2ServerAutoConfiguration.class,
    MockLoginController.class
})
@PropertyMapping(MockOAuth2ServerProperties.PREFIX)
public @interface EnableMockOAuth2Server {
    /**
     * Specify port for server to run on (only works in test scope), provide via
     * env property mock-ouath2-server.port outside of test scope
     */
    int port() default 0;
}

File: token-validation-spring-test/src/main/kotlin/no/nav/security/token/support/spring/test/EnableMockOAuth2Server.kt
package no.nav.security.token.support.spring.test

import org.springframework.boot.test.context.PropertyMapping
import java.lang.annotation.Inherited
import org.springframework.context.annotation.Import
import kotlin.annotation.AnnotationRetention.RUNTIME
import kotlin.annotation.AnnotationTarget.CLASS

@MustBeDocumented
@Inherited
@Retention(RUNTIME)
@Target(CLASS)
@Import(MockOAuth2ServerAutoConfiguration::class, MockLoginController::class)
@PropertyMapping(MockOAuth2ServerProperties.PREFIX)
annotation class EnableMockOAuth2Server(
    /**
     * Specify port for server to run on (only works in test scope), provide via
     * env property mock-ouath2-server.port outside of test scope
     */
    val port : Int = 0)